libraries/rush-lib/src/utilities/prompts/SearchListPrompt.ts

```diff
@@ -0,0 +1,278 @@
+  private _firstRender: boolean = true;
+
+  public constructor(question: ListQuestion, readline: Interface, answers: Answers) {
+    super(question, readline, answers);
+
+    if (!this.opt.choices) {
+      this.throwParamError('choices');
+    }
+
+    if (
+      _.isNumber(this.opt.default) &&
+      this.opt.default >= 0 &&
+      this.opt.default < this.opt.choices.realLength
+    ) {
+      this._selected = this.opt.default;
+    } else if (!_.isNumber(this.opt.default) && this.opt.default !== null) {
+      const index: number = _.findIndex(
+        this.opt.choices.realChoices,
+        ({ value }) => value === this.opt.default
+      );
+      this._selected = Math.max(index, 0);
+    }
+
+    // Make sure no default is set (so it won't be printed)
+    this.opt.default = null;
+
+    this._paginator = new Paginator(this.screen);
+  }
+
+  protected _run(callback: (result: unknown) => void): this {
+    this.done = callback;
+
+    // eslint-disable-next-line @typescript-eslint/typedef
+    const events = observe(this.rl);
+    // eslint-disable-next-line @typescript-eslint/typedef
+    const validation = this.handleSubmitEvents(events.line.pipe(map(this._getCurrentValue.bind(this))));
+
+    //eslint-disable-next-line no-void
+    void validation.success.forEach(this._onSubmit.bind(this));
+    //eslint-disable-next-line no-void
+    void validation.error.forEach(this._onError.bind(this));
+
+    // eslint-disable-next-line no-void
+    void events.numberKey
+      .pipe(takeUntil(events.line))
+      .forEach(this._onNumberKey.bind(this) as (evt: unknown) => void);
+
+    // eslint-disable-next-line no-void
+    void events.keypress
+      .pipe(takeUntil(validation.success))
+      .forEach(this._onKeyPress.bind(this) as (evt: unknown) => void);
+
+    this.render();
+    return this;
+  }
+
+  private _onUpKey(): void {
+    return this._adjustSelected(-1);
+  }
+
+  private _onDownKey(): void {
+    return this._adjustSelected(1);
+  }
+
+  private _onNumberKey(input: number): void {
+    if (input <= this.opt.choices.realLength) {
+      this._selected = input - 1;
+    }
+
+    this.render();
+  }
+
+  /**
+   * When user press `enter` key
+   */
+  private _onSubmit(state: { value: unknown }): void {
+    this.status = 'answered';
+    // Rerender prompt (and clean subline error)
+    this.render();
+
+    this.screen.done();
+    this.done(state.value);
+  }
+
+  private _onError(state: inquirer.prompts.FailedPromptStateData): void {
+    this.render(state.isValid || undefined);
+  }
+
+  private _onKeyPress(event: IKeyPressEvent): void {
+    if (event.key.ctrl) {
+      switch (event.key.name) {
+        case 'backspace':
+          return this._setQuery('');
+      }
+    } else {
+      switch (event.key.name) {
+        case 'backspace':
+          return this._setQuery(this._query.slice(0, -1));
+        case 'up':
+          return this._onUpKey();
+        case 'down':
+          return this._onDownKey();
```

Comment:
home, end, pageup and pagedown are all quite handy for navigating large lists

Reply:
Sounds good we can add these back in.